kubernetes/structures_deployment.go

```diff
@@ -75,7 +77,13 @@ func expandDeploymentSpec(deployment []interface{}) (*appsv1.DeploymentSpec, err
 	obj.MinReadySeconds = int32(in["min_ready_seconds"].(int))
 	obj.Paused = in["paused"].(bool)
 	obj.ProgressDeadlineSeconds = ptrToInt32(int32(in["progress_deadline_seconds"].(int)))
-	obj.Replicas = ptrToInt32(int32(in["replicas"].(int)))
+	if v, ok := in["replicas"].(string); ok && v != "" {
+		i, err := strconv.Atoi(v)
+		if err != nil {
+			return obj, err
+		}
+		obj.Replicas = ptrToInt32(int32(i))
```

Comment:
There's a CodeQL warning here because the return type of Atoi() will be at least an _int32_ meaning it can be bigger on some architectures. In this case, type-casting to int32 might be lossy and lead to unexpected values. The recommendation here is to check against the ceiling value of int32 before conversion.

Realistically, this is highly unlikely in the case of a replicas value. But automation gone wrong or a mistake could certainly cause it.

There's one more such warning for stateful sets.

Reply:
Have updated to use ParseInt with a bitsize of 32 so that we get an error if it doesn't fit.